internal/internal.go

```diff
@@ -391,5 +391,32 @@ func parseTime(format string, timestamp string, location *time.Location) (time.T
 	case "stampnano":
 		format = time.StampNano
 	}
-	return time.ParseInLocation(format, timestamp, loc)
+
+	if !strings.Contains(format, "MST") {
+		return time.ParseInLocation(format, timestamp, loc)
+	}
+
+	// Golang does not parse times with ambiguous timezone abbreviations,
+	// but only parses the time-fields and the timezone NAME with a zero
+	// offset (see https://groups.google.com/g/golang-nuts/c/hDMdnm_jUFQ/m/yeL9IHOsAQAJ).
+	// To handle those timezones correctly we can use the timezone-name and
+	// force parsing the time in that timezone. This way we get the correct
+	// time for the "most probably" of the ambiguous timezone-abbreviations.
```

Comment:
What does this mean in practice? For example, I live in MST/MDT. What happens if the logs I am parsing are actually from Malaysian standard time (the other MST).

Reply:
It means that zones like "MST" and "PST" are "Mountain standard" and "Pacific standard" as you probably would expect. Of you want to parse "Malaysian standard time" and use "MST" you are doomed. :-) However, I don't see any other way except for we have a way for the user to provide an _extensive_ (i.e. covering all abbreviations that _might_ occur) mapping of abbrev to time-offset...